File: loki_client.py
#!/usr/bin/env python3

# Usage: ./loki_client.py /path/to/scan

import os
import socket
import sys, time

host = '127.0.0.1'
port = 1337

try:
    path = sys.argv[1]
except:
    print('missing path')
    sys.exit(1)

try:
    auth = sys.argv[2]
except:
    auth = ''

if not os.path.isfile(path) and not os.path.isdir(path):
    print(path + ' not found')
    sys.exit(127)

try:
    client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
except socket.error:
    print('Could not create a socket')
    sys.exit()

try:
    client.connect((host, port))
except socket.error:
    print('Could not connect to server')
    sys.exit()

if auth:
    data = sys.argv[1] + " " + sys.argv[2]
else:
    data = sys.argv[1]

client.send(data.encode())
message = client.recv(2048)
print(message.decode())
client.close()

File: loki_client.php
<?php

$host = "127.0.0.1";
$port = 1337;

$path = $argv[1];

if(isset($argv[2])) {
    $auth = $argv[2];
    $message = $path . " " . $auth;
} else {
    $message = $path;
}

if(!is_file($path) and !is_dir($path))
{
    die("path: $message doesn't exist\n");
}

$socket = @socket_create(AF_INET, SOCK_STREAM, 0);
$result = @socket_connect($socket, $host, $port);

if($result) {
    socket_write($socket, $message, strlen($message)) or die("Could not send data to server\n");
    $result = socket_read ($socket, 2048) or die("Could not read server response\n");
    echo $result;
} else {
    echo 'not connected';
}

echo "\n";

socket_close($socket);

?>
